platform/platform-logging/src/main/java/org/codice/ddf/logging/platform/LoggingServiceBean.java

```diff
@@ -0,0 +1,200 @@
+                LOGGER.info("Registered Logging Service MBean under object name: {}",
+                        objectName.toString());
+            } catch (InstanceAlreadyExistsException e) {
+                mBeanServer.unregisterMBean(objectName);
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Re-registered Logging Service MBean");
+            }
+        } catch (MBeanRegistrationException | InstanceNotFoundException
+                | InstanceAlreadyExistsException | NotCompliantMBeanException e) {
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    public void destroy() {
+        try {
+            if (objectName != null && mBeanServer != null) {
+                mBeanServer.unregisterMBean(objectName);
+                LOGGER.info("Unregistered Logging Service MBean");
+            }
+        } catch (InstanceNotFoundException | MBeanRegistrationException e) {
+            LOGGER.error("Exception unregistering MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    /**
+     * Called each time a {@link org.ops4j.pax.logging.spi.PaxLoggingEvent} is created in the system
+     */
+    @Override
+    public void doAppend(PaxLoggingEvent paxLoggingEvent) {
+        LogEvent logEvent = createLogEvent(paxLoggingEvent);
+        add(logEvent);
+    }
+
+    @Override
+    public List<LogEvent> retrieveLogEvents() {
+        try {
+            readLock.lock();
+            return Lists.newArrayList(logEvents);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the maximum number of {@link LogEvent}s to store
+     * 
+     * @param newMaxLogEvents
+     *            This number cannot be less than 0 or greater than {@code MAX_LOG_EVENTS_LIMIT}. In
+     *            the event that this parameter is less than 0 or greater than
+     *            {@code MAX_LOG_EVENTS_LIMIT}, the maximum log events stored will set set to
+     *            {@code MAX_LOG_EVENTS_LIMIT}.
+     */
+    public void setMaxLogEvents(int newMaxLogEvents) {
+        try {
+            writeLock.lock();
+            if (newMaxLogEvents <= 0 || newMaxLogEvents > MAX_LOG_EVENTS_LIMIT) {
+                LOGGER.warn(
+                        "An invalid value of [{}] was entered for maximum log events to store. This "
+                                + "value must be greater than 0 and must not exceed [{}]. Maximum log"
+                                + " events to store will be set to [{}].", newMaxLogEvents,
+                        MAX_LOG_EVENTS_LIMIT, DEFAULT_LOG_EVENTS_LIMIT);
+                newMaxLogEvents = DEFAULT_LOG_EVENTS_LIMIT;
+            }
+
+            EvictingQueue<LogEvent> evictingQueue = EvictingQueue.create(newMaxLogEvents);
+
+            if (logEvents.size() < newMaxLogEvents) {
+                evictingQueue.addAll(logEvents);
+            } else {
```

Comment:
Like we talked, you shouldn't need the conditional here. You should be able to simply call `addAll()` on the new queue and let it evict the extra ones for you if the new list is smaller.